docs/src/main/asciidoc/feature-management.adoc

```diff
@@ -0,0 +1,279 @@
+              rolloutPercentage: 100
+            -
+              name: Ring1
+              rolloutPercentage: 100
+          defaultRolloutPercentage: 50
+----
+
+=== Custom Feature Filters
+
+Creating a feature filter provides a way to enable features based on criteria that you define. To implement a feature filter, the `FeatureFilter` interface must be implemented. `FeatureFilter` has a single method `evaluate`. When a feature specifies that it can be enabled with a feature filter, the `evaluate` method is called. If `evaluate` returns `true` it means the feature should be enabled. If `false` it will continue evaluating the Feature's filters until one returns true. If all return `false` then the feature is off.
+
+Feature filters are found by being defined as being Spring Beans, so they are either defined as `@Component` or defined in an `@Configuration`.
+
+[source,java,indent=0]
+----
+@Component("Random")
+public class Random implements FeatureFilter {
+
+    @Override
+    public boolean evaluate(FeatureFilterEvaluationContext context) {
+        double chance = Double.valueOf((String) context.getParameters().get("chance"));
+        return Math.random() > chance / 100;
+    }
+
+}
+----
+
+==== Parameterized Feature Filters
+
+Some feature filters require parameters to decide whether a feature should be turned on or not. For example, a browser feature filter may turn on a feature for a certain set of browsers. It may be desired that Edge and Chrome browsers enable a feature, while Firefox does not. To do this, a feature filter can be designed to expect parameters. These parameters would be specified in the feature configuration and in code, and would be accessible via the `FeatureFilterEvaluationContext` parameter of `evaluate`. `FeatureFilterEvaluationContext` has a property `parameters` which is a `HashMap<String, Object>`.
+
+=== Targeting
+
+Targeting is a feature management strategy that enables developers to progressively roll out new features to their user base. The strategy is built on the concept of targeting a set of users known as the target audience. An audience is made up of specific users, groups, and a designated percentage of the entire user base. The groups that are included in the audience can be broken down further into percentages of their total members.
+
+The following steps demonstrate an example of a progressive rollout for a new 'Beta' feature:
+
+1. Individual users Jeff and Alicia are granted access to the Beta
+2. Another user, Mark, asks to opt-in and is included.
+3. Twenty percent of a group known as "Ring1" users are included in the Beta.
+4. The number of "Ring1" users included in the beta is bumped up to 100 percent.
+5. Five percent of the user base is included in the beta.
+6. The rollout percentage is bumped up to 100 percent and the feature is completely rolled out.
+7. This strategy for rolling out a feature is built into the library through the included TargetingFilter feature filter.
```

Comment:
Really really like this list example - only , you do not need to mention the final description line as step 7, should just be on a new line